dali/pipeline/executor/executor.cc

```diff
@@ -524,6 +530,12 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunHelper(OpNode &op_node, Workspac
       in.SetLayout({});
     }
   }
+
+  // If it is the end of an epoch, create a checkpoint
+  if (checkpointing_ && (iteration_id + 1) % checkpointing_epoch_size_ == 0) {
```

Comment:
If `checkpointing_epoch_size_ == 1` we end up doing two checkpoints in the `iteration_id == 0`, right?

Reply:
No. The here the checkpoint created is corresponding to the state between epochs `(iterations_id, iteration_id + 1)`. The checkpoint created before `iteration_id == 0` is something else.

I will try to improve the comment here to reflect that.